### DISCUSSION TITLE
Is it possible to use block_images with undetectable?

### DISCUSSION THREAD

**Discussion by MalvinJohnas** [seleniumbase/SeleniumBase#2198] [Q&A] [answered]
I am building a web scraper for a marketplace, and downloading images for each product takes a lot of bandwidth. Therefore, I need to find a way to block downloading images or to download images to the browser cache and then reuse them, so I don't need to download them each time I visit the page.

I've tried using the following code; however, it doesn't work when block_images is set to true.

```
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class CoffeeCartTest(BaseCase):
    def test_chrome(self):
        self.get_new_driver(undetectable=True, headless=False, block_images=True)
        self.open("https://nowsecure.nl/")
        input('abc')
        self.quit
```

**Comment by mdmintz** [comment, marked as answer]
Blocking images will make you detectable, but if you reuse a custom `user_data_dir` for each run, you won't have to reload certain resources that have already been cached.

https://github.com/seleniumbase/SeleniumBase/blob/be6ab2a4a999a5c26a942c99d6064756cecbdc48/help_docs/method_summary.md?plain=1#L301

(`user_data_dir` is one of the options in the `get_new_driver()` method)
